packages/utils/src/conversions.ts

```diff
@@ -220,3 +220,49 @@ export function bufferToHexadecimal(value: Buffer | Uint8Array): string {
 
     return Buffer.from(value).toString("hex")
 }
+
+/**
+ * Converts a buffer to a base64 string. It accepts 'Buffer' or 'Uint8Array'.
+ * @param value The buffer to convert.
+ * @returns The converted base64 string.
+ */
+export function bufferToBase64(value: Buffer | Uint8Array): string {
+    requireTypes(value, "value", ["Buffer", "Uint8Array"])
+
+    return Buffer.from(value).toString("base64")
+}
+
+/**
+ * Converts a base64 string to a buffer. This function does not check
+ * if the input value is a valid base64 string.
+ * @param value The base64 string to convert.
+ * @returns The converted buffer.
+ */
+export function base64ToBuffer(value: string): Buffer {
+    requireString(value, "value")
+
+    return Buffer.from(value, "base64")
+}
+
+/**
+ * Converts text (utf8) to a base64 string.
+ * @param value The text to convert.
+ * @returns The converted base64 string.
+ */
+export function textToBase64(value: string): string {
+    requireString(value, "value")
+
+    return Buffer.from(value, "utf8").toString("base64")
```

Comment:
I'm not confident enough in my knowledge of Unicode in JavaScript to be sure this is the right way to deal with non-ASCII text.  I'd suggest some unit tests which exercise this with some non-ASCII text, including multilingual stuff (accents, cyrilic, Chinese) and some emojis, and see that they round-trip correctly through `textToBase64` -> `base64ToText`.

Reply:
`utf-8` supports all characters in the Unicode standard and it's the preferred encoding for web content. I tested many different characters and they all work 👍🏽

Reply:
Agreed, I just wasn't sure what the semantics of JavaScript strings were in that regard.  In some languages, strings separate their internal encoding from external encodings.

One case to consider documenting here is what happens to invalid utf8 sequences in the input.  I presume they'll probably get ignored similarly to the invalid base64 in other cases.

Reply:
Can you make an example? An input like "��" will be correctly converted.

Reply:
Funny.  I don't know if you really meant to send question-mark charaters or if GitHub + Chrome isn't decoding something.
![image](https://github.com/privacy-scaling-explorations/zk-kit/assets/1761206/3a2c7a06-d5a9-427b-af44-0c61b47edef1)

I was thinking of strings which contain non-UTF data.  Not every possible sequence of bytes is valid UTF8.  Look at the encding [here](https://en.wikipedia.org/wiki/UTF-8#Encoding) and you can see if the first byte indicates a multi-byte sequence, then the remaining bytes must have high bits 10, and if they don't it would be an invalid encoding.

It may not be worth worrying about, as bad utf8 is much less of a concern than bad base64.  The structure of the code simply makes me think about any decoding being subject to failure.  I think it's fine that the answer here is "we do what JavaScript does" or "we do what Buffer does".  The main bit of clarification I'd be looking for as a user would be whether the result is an exception, or the result is truncated output.

Reply:
I actually sent two question marks, but that's a character that should be correctly supported actually right? :D